### ISSUE TITLE
KVO: NSKeyValueObservingOptionInitial issue when observing multiple nested properties

### ISSUE THREAD

**Issue by triplef** [Microsoft/WinObjC#226]
##### Summary

When observing multiple nested properties, each `addObserver:` call using `NSKeyValueObservingOptionInitial` will also trigger change notifications for every key path with an identical first key that is already being observed. When using `NSKeyValueObservingOptionNew`, the "new" object in the change dictionary will always be the value of the first key path that was observed with the same first key.

##### Example

Assuming `object` has a property `child` returning an object with the properties `property1`,  `property2`, and  `property3`, the following code causes 5 change notifications:

```obj-c
[object addObserver:self forKeyPath:@"child.property1" options:NSKeyValueObservingOptionNew|NSKeyValueObservingOptionInitial context:nil];
[object addObserver:self forKeyPath:@"child.property2" options:NSKeyValueObservingOptionNew|NSKeyValueObservingOptionInitial context:nil];
[object addObserver:self forKeyPath:@"child.property3" options:NSKeyValueObservingOptionNew|NSKeyValueObservingOptionInitial context:nil];
```

1. The first line will trigger a change notification for `child.property1`.
2. The second line will trigger a change notification for `child.property1` and `child.property2`.
3. The third line will trigger a change notification for `child.property1`, `child.property2`, and `child.property3`.

The reported new value in the change notification will be the value of "child.property1" for all 5 change notifications.

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)
